Classes/Tools.php

```diff
@@ -33,7 +33,7 @@
  *
  * @license http://www.gnu.org/licenses/gpl.html GNU General Public License, version 3 or later
  */
-abstract class Tools
+class Tools
```

Comment:
You can do this, but it is inefficient, it just means the PHP has to parse another file. It is more efficient to not set it as abstract.  You use the Tools class throughout your script even though not once have I seen this abstract extended to another class as all the methods are static. So there is no reason for the class to be an abstract.